service/frontend/workflowHandler.go

```diff
@@ -766,19 +766,22 @@ func (wh *WorkflowHandler) PollWorkflowTaskQueue(ctx context.Context, request *w
 
 	err = backoff.Retry(op, frontendServiceRetryPolicy, common.IsServiceTransientError)
 	if err != nil {
-		errCancel := wh.cancelOutstandingPoll(ctx, err, namespaceID, enumspb.TASK_QUEUE_TYPE_WORKFLOW, request.TaskQueue, pollerID)
-		if errCancel != nil {
-			// For all other errors log an error and return it back to client.
-			ctxTimeout := "not-set"
-			ctxDeadline, ok := ctx.Deadline()
-			if ok {
-				ctxTimeout = ctxDeadline.Sub(callTime).String()
-			}
-			wh.GetLogger().Error("PollWorkflowTaskQueue failed.",
-				tag.WorkflowTaskQueueName(request.GetTaskQueue().GetName()),
-				tag.Value(ctxTimeout),
-				tag.Error(errCancel))
+		err = wh.cancelOutstandingPoll(ctx, err, namespaceID, enumspb.TASK_QUEUE_TYPE_WORKFLOW, request.TaskQueue, pollerID)
```

Comment:
To prevent confusion in the future, let's have `cancelOutstandingPoll` return `bool` instead of `error`. That will eliminate rewriting of `err` and streamline the logic here.